src/coprocessor/codec/mysql/time/mod.rs

```diff
@@ -242,30 +276,44 @@ impl Time {
         }
     }
 
+    fn split_datetime(s: &str) -> (Vec<&str>, &str) {
+        let trimmed = s.trim();
+        if trimmed.is_empty() {
+            return (vec![], "");
+        }
+        let (parts, fracs) = if let Some(i) = trimmed.rfind('.') {
+            (&trimmed[..i], &trimmed[i + 1..])
+        } else {
+            (trimmed, "")
+        };
+        (Time::parse_datetime_format(parts), fracs)
+    }
+
     pub fn parse_utc_datetime(s: &str, fsp: i8) -> Result<Time> {
         Time::parse_datetime(s, fsp, Tz::utc())
     }
 
     pub fn parse_datetime(s: &str, fsp: i8, tz: Tz) -> Result<Time> {
         let fsp = mysql::check_fsp(fsp)?;
-        let mut frac_str = "";
         let mut need_adjust = false;
-        let parts = Time::parse_datetime_format(s);
+        let mut has_hhmmss = false;
```

Comment:
Would you also like to add a isFloat option? https://github.com/pingcap/tidb/blob/9cf670a32493c3cc36a7f9d9bc5c7e91953d0d78/types/time.go#L603

Reply:
ok